core/src/main/java/org/apache/iceberg/ManifestFilterManager.java

```diff
@@ -372,8 +367,14 @@ private boolean manifestHasDeletedFiles(
 
     for (ManifestEntry<F> entry : reader.liveEntries()) {
       F file = entry.file();
+
+      // add path-based delete to set of files to be deleted
+      if (deletePaths.contains(CharSequenceWrapper.wrap(file.path()))) {
```

Comment:
Why do we wrap? It is `CharSequenceSet`.

Reply:
oops good point, I didn't pay enough attention here and just re-used what the old impl was doing in https://github.com/apache/iceberg/blob/main/core/src/main/java/org/apache/iceberg/ManifestFilterManager.java#L444-L445